meta/store_test.go

```diff
@@ -608,6 +608,37 @@ func TestStore_UserCount(t *testing.T) {
 	}
 }
 
+// Ensure the store can take a snapshot.
+func TestStore_Snapshot_And_Restore(t *testing.T) {
+	t.Parallel()
+
+	s := MustOpenStore()
+	s.LeaveFiles = true
+
+	// Test taking a snapshot.
+	if err := s.Store.Snapshot(); err != nil {
+		t.Fatal(err)
+	}
+	s.Close()
+
+	// Test restoring the snapshot taken above.
+	existingDataPath := s.Path()
+	s = NewStore(NewConfig(existingDataPath))
+	if err := s.Open(); err != nil {
+		panic(err)
+	}
+	defer s.Close()
+
+	// Wait until the server is ready.
+	select {
+	case err := <-s.Err():
+		panic(err)
+	case <-s.Ready():
+	}
+
+	// TODO: figure out some way to confirm that the snapshot restored
+}
```

Comment:
This function should write a lot of changes to the `meta.Store` to force a snapshot and then restart and verify that everything is there. Maybe do a bunch of `Store.CreateDatabase()` calls. I don't think calling `Snapshot()` is enough coverage to make sure it works with the `hashicorp/raft` library.

You can also lower the `raft.Config.SnapshotInterval` and `raft.Config.SnapshotThreshold` to ensure snapshots occur more frequently.